test/Tests/src/Data/Time/Date_Time_Spec.enso

```diff
@@ -323,6 +323,45 @@ spec_with name create_new_datetime parse_datetime nanoseconds_loss_in_precision=
+            time+Date_Period.Week . should_equal <| create_new_datetime 1970 1 8 (zone = Time_Zone.utc)
+            time+Date_Period.Day . should_equal <| create_new_datetime 1970 1 2 (zone = Time_Zone.utc)
+
+        Test.specify "should support subtraction of Date_Period" <|
+            time = create_new_datetime 1970 (zone = Time_Zone.utc)
+            time-Date_Period.Year . should_equal <| create_new_datetime 1969 (zone = Time_Zone.utc)
+            time-Date_Period.Quarter . should_equal <| create_new_datetime 1969 10 (zone = Time_Zone.utc)
+            time-Date_Period.Month . should_equal <| create_new_datetime 1969 12 (zone = Time_Zone.utc)
+            time-Date_Period.Week . should_equal <| create_new_datetime 1969 12 25 (zone = Time_Zone.utc)
+            time-Date_Period.Day . should_equal <| create_new_datetime 1969 12 31 (zone = Time_Zone.utc)
+
+        Test.specify "should support addition of Time_Period" <|
+            time = create_new_datetime 1970 (zone = Time_Zone.utc)
+            time+Time_Period.Day . should_equal <| create_new_datetime 1970 1 2 (zone = Time_Zone.utc)
+            time+Time_Period.Hour . should_equal <| create_new_datetime 1970 1 1 1 (zone = Time_Zone.utc)
+            time+Time_Period.Minute . should_equal <| create_new_datetime 1970 1 1 0 1 (zone = Time_Zone.utc)
+            time+Time_Period.Second . should_equal <| create_new_datetime 1970 1 1 0 0 1 (zone = Time_Zone.utc)
+
+        Test.specify "should support subtraction of Time_Period" <|
+            time = create_new_datetime 1970 (zone = Time_Zone.utc)
+            time-Time_Period.Day . should_equal <| create_new_datetime 1969 12 31 (zone = Time_Zone.utc)
+            time-Time_Period.Hour . should_equal <| create_new_datetime 1969 12 31 23 (zone = Time_Zone.utc)
+            time-Time_Period.Minute . should_equal <| create_new_datetime 1969 12 31 23 59 (zone = Time_Zone.utc)
+            time-Time_Period.Second . should_equal <| create_new_datetime 1969 12 31 23 59 59 (zone = Time_Zone.utc)
+
+        Test.specify "should support mixed addition and subtraction of Date_Period and Time_Period" <|
+            time = create_new_datetime 1970 (zone = Time_Zone.utc)
+            time+Date_Period.Quarter+Time_Period.Hour . should_equal <| create_new_datetime 1970 4 1 1 (zone = Time_Zone.utc)
+            time+Time_Period.Hour+Date_Period.Quarter . should_equal <| create_new_datetime 1970 4 1 1 (zone = Time_Zone.utc)
+            time+Date_Period.Year-Time_Period.Minute . should_equal <| create_new_datetime 1970 12 31 23 59 (zone = Time_Zone.utc)
+            time+Time_Period.Minute+Time_Period.Minute-Time_Period.Minute . should_equal <| create_new_datetime 1970 1 1 0 1 (zone = Time_Zone.utc)
+            time+Date_Period.Quarter-Date_Period.Month . should_equal <| create_new_datetime 1970 3 1 (zone = Time_Zone.utc)
+            time+Date_Period.Day-Time_Period.Day . should_equal <| create_new_datetime 1970 (zone = Time_Zone.utc)
```

Comment:
(As for documenting the difference, I guess this raises another question - even if we add the docs to the constructors - they will only be displayed when searched in Component Browser. That is already good. But most of the time users will be picking these using the dropdowns; and then it will actually not be that easy to discover this documentation. This leads to a possible longer-term direction to explore - shall dropdowns somehow "connect" to documentation? IMO ability to access documentation is very important to be able to understand the product; this is not for the nearest term, but I think it's worth thinking how we can connect these. Maybe we can display docs on hover over an option? Or have some question mark signs that could open the doc panel? cc: @jdunkerley)

Reply:
No question, we will need to improve how docs for args are displayed but will be a while until we get there.

Reply:
Created https://github.com/enso-org/enso/issues/6969 for documentation.

Reply:
Ok, but will appreciate if the test can be added as part of this PR so we have this behaviour documented.

Reply:
Sorry, hadn't pushed -- it's there now.